types/index.d.ts

```diff
@@ -70,7 +70,9 @@ export interface RenderOptions<
    *
    *  @see https://testing-library.com/docs/react-testing-library/api/#wrapper
    */
-  wrapper?: React.ComponentType
+  wrapper?:
+    | React.ComponentType
+    | ((props: {children: React.ReactNode}) => JSX.Element)
```

Comment:
Let me add a type test for this because the fix is simpler I believe.

Reply:
Implicit children strikes again:

`React.ComponentType` includes `React.FunctionComponent` which means the accepted props are `{ children?: React.ReactNode }`. However, our `Wrapper` only accepts `{ children: React.ReactElement }` props. So we could render `<Wrapper>Some Child</Wrapper>` which may not be implemented by the custom `Wrapper` component because now it receives `{ children: string }`.

The proposed fix would not have fixed the linked issue (`npm typecheck` fails on ffa222170ab0f4654319e611b5f48fce8590ddbf). dc03098550817b9e222473c97fdc40773847c05f fixes the linked issue.